sentry-android-core/src/main/java/io/sentry/android/core/NdkIntegration.java

```diff
@@ -26,6 +26,12 @@ public final void register(final @NotNull IHub hub, final @NotNull SentryOptions
     Objects.requireNonNull(hub, "Hub is required");
     Objects.requireNonNull(options, "SentryOptions is required");
 
+    final String cachedDir = options.getCacheDirPath();
+    if (cachedDir == null || cachedDir.isEmpty()) {
+      options.getLogger().log(SentryLevel.WARNING, "No cache dir path is defined in options.");
```

Comment:
it depends the point of view, as we add the integration and the flag is enabled by default, it'd make sense to be an error, but having a caching folder is optional anyway, so I'd say it's a warning, if the user disables `isEnabledSdk`, then this won't be even ran

Reply:
The cache folder is optional if you don't use the NDK integration. Otherwise it'll be broken so I'd say it makes sense to log error if we have NDK + no cache path.

Reply:
it's not optional, this PR makes it optional.